src/groupeddataframe/fastaggregates.jl

```diff
@@ -198,11 +199,15 @@ function groupreduce!(res::AbstractVector, f, op, condf, adjust, checkempty::Boo
     if checkempty && any(iszero, counts)
         throw(ArgumentError("some groups contain only missing values"))
     end
-    # Undo pool sharing done by groupreduce_init
-    if res isa CategoricalVector && res.pool === incol.pool
-        V = Union{CategoricalArrays.leveltype(res),
-                  eltype(res) >: Missing ? Missing : Union{}}
-        res = CategoricalArray{V, 1}(res.refs, copy(res.pool))
+    # Reallocate Vector created in groupreduce_init with min or max
+    # for CategoricalVector
+    incolT = typeof(incol).name
+    if incolT.name === :CategoricalArray &&
+        nameof(incolT.module) === :CategoricalArrays && res isa Vector
```

Comment:
Why the `res isa Vector` check?
```suggestion
    resT = typeof(res).name
    if resT.name === :CategoricalArray &&
        nameof(resT.module) === :CategoricalArrays
```